district_data_import.py

```diff
@@ -233,6 +244,12 @@
                     dp_donorrecord['EMAIL'] = district_record['Parent1Email']
                 elif district_record['Parent2Email'] and dp_donorrecord['FIRST_NAME'] == district_record['Parent 2 First Name']:
                     dp_donorrecord['EMAIL'] = district_record['Parent2Email']
+            #set homeschool if needs to be changed by looking at all students for the donor
+            all_students = dp.get_students_for_donor(dp_studentrecord['DONOR_ID'])
+            old_homeschool = dp_donorrecord['HOME_SCHOOL']
+            dp_donorrecord['HOME_SCHOOL'] = dp.calculate_homeschool(all_students)
+            if not dp_donorrecord['FORMER_ELEM_SCHOOL'] and dp_donorrecord['HOME_SCHOOL'] == 'BIS':
+                dp_donorrecord['FORMER_ELEM_SCHOOL'] = '' if old_homeschool == 'BIS' else old_homeschool
```

Comment:
I no longer remember why multi-donor students have a different branch, but it sure looks like a lot of the logic is missing on that side. Do you know how many of these there are?

Is it possible to avoid having the same code in both branches?

Reply:
We may not have even a dozen students who belong to multiple donor records.  Very minor cases.  But yes, we are missing some of the logic in there -- should probably fix that.

Reply:
I take it back -- we have maybe two -three dozen students with multiple donors.